addons/mail/controllers/discuss.py

```diff
@@ -196,6 +196,8 @@ def discuss_starred_messages(self, max_id=None, min_id=None, limit=30, **kwargs)
 
     @http.route('/mail/message/post', methods=['POST'], type='json', auth='public')
     def mail_message_post(self, thread_model, thread_id, post_data, **kwargs):
+        attachment_ids = post_data.get('attachment_ids', [])
+        request.env['mail.thread']._message_post_check_attachments(attachment_ids=attachment_ids, attachment_tokens=post_data.get('attachment_tokens', [None] * len(attachment_ids)))
```

Comment:
why the `[None] * len(attachment_ids)` ? The check is going to fail anyway if there is not tokens.

Reply:
Tokens are only checked if the current user does not naturally have access to the document, which could be the case for internal users. (cf. the methods that have been moved from portal)

This ensures the most possible compatibility (running the old JS on the new server, or the old server with the new JS). In most situations it will just keep working like before, and only for guest both need to be fully updated.